src/main/host/descriptor/mod.rs

```diff
@@ -139,6 +139,15 @@ impl From<FileState> for c::Status {
     }
 }
 
+bitflags::bitflags! {
+    /// File-related signals that listeners can watch for.
+    #[derive(Default, Copy, Clone, Debug)]
+    #[repr(transparent)]
+    pub struct FileSignals: u32 {
```

Comment:
Are you sure you want to make this a plural?

Reply:
This is plural since the bitflags type represents a collection of signals. For example if we had `pub struct FileSignals: u32 { const A = 1, const B = 2 }`, then `A` and `B` would each be an individual signal, and `let foo: FileSignals = A | B` would be a collection of both signals `A` and `B`. Combining two independent signals doesn't make them a single signal (at least in the case of Shadow).

The `FileState` bitflags type is not plural since the state itself is a collection of flags. For example if a file is readable and writable (`READABLE | WRITABLE`), then both of those flags make up the file's "state". Maybe it could have been named `FileStateFlags` instead which would have been plural, and then the "state" would be a collection of "flags".